subsys/bluetooth/host/smp.c

```diff
@@ -2775,6 +2775,11 @@ static u8_t smp_pairing_req(struct bt_smp *smp, struct net_buf *buf)
 	    (req->auth_req & BT_SMP_AUTH_BONDING)) {
 		atomic_set_bit(smp->flags, SMP_FLAG_BOND);
 	}
+#if defined(CONFIG_BT_BONDING_REQUIRED)
+    else {
+        return BT_SMP_ERR_AUTH_REQUIREMENTS;
+    }
+#endif /* CONFIG_BT_BONDING_REQUIRED */
```

Comment:
```
        } else if (IS_ENABLED(CONFIG_BT_BONDING_REQUIRED) {
                return BT_SMP_ERR_AUTH_REQUIREMENTS
```

Don't we need to cover for all the other scenarios as well, i.e. receiving pairing response or security request?

Reply:
This will reject security elevation as well. Not sure what you mean by the pairing response part?

Reply:
Depending on our role we discover the remote authentication requirements by receiving one of the following PDUs from the peer: Pairing Request, Pairing Response, Security Request. It seems your PR only covers the first one.

Reply:
The central will receive these:
 - Security Request (if initiated by the peripheral)
 - Pairing Response (always)

Peripheral receive this:
 - Pairing Request

@demantmrrv With your change now you have added support for this feature for the peripheral. To get central support you need the central received ones.
Security elevation would still be supported, as long as the bond flag is set.

I think returning ERR_AUTH_REQ might send the wrong information back to the peer. It might suggest that stronger authentication is required.
```
The pairing procedure cannot be per-
formed as authentication requirements
cannot be met due to IO capabilities of one
or both devices
```

I'm not 100% sure what is the best error to give to, so maybe just "UNSPECIFIED" is best, but maybe "CMD_NOT_SUPPORTED" would work?

Reply:
Ah, sorry, I'm a Zephyr noob. I can see the resemblance in smp_pairing_rsp and I need to add code in smp_security_request as well. The reason code is difficult as the spec is (once again) a little weak on the subject. I would probably go with the Unspecified Reason, as Command Not Supported seems a bit harsh. Conversely, the command is practically not supported in this specific case, but available in general. IDK.

Reply:
I'd say the command is supported but the parameters are unacceptable, so I wouldn't use Command Not Supported.

Reply:
for parameters we only have `BT_SMP_ERR_INVALID_PARAMS` but that would mean that the parameters are outside of valid ranges. So i wouldn't use that.
I think we should use UNSPECIFIED then.

What is the peer supposed to do in the case where the pairing is rejected?  I suppose it doesn't know, and therefore is not part of the system. So the unspecified is acceptable? There is no action to take.

Reply:
@joerchan I concur. In this case the device cannot accept the AuthReq parameters as is, but they are not invalid per se. Exemption leave us Unspecified Reason.

Reply:
@joerchan @jhedberg I added the code to the central role. I was reluctant to use the bondable variable directly, but realized it is indirectly used the other places through get_auth() anyway.